library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -137,32 +134,34 @@ class ChuckerInterceptor @JvmOverloads constructor(
     /**
      * Processes a [ResponseBody] and populates corresponding fields of a [HttpTransaction].
      */
-    private fun processResponseBody(response: Response, responseBody: ResponseBody?, transaction: HttpTransaction) {
-        getNativeSource(response)?.use { source ->
-            source.request(java.lang.Long.MAX_VALUE)
-            val buffer = source.buffer()
-            var charset: Charset = UTF8
-            val contentType = responseBody?.contentType()
-            if (contentType != null) {
-                try {
-                    charset = contentType.charset(UTF8) ?: UTF8
-                } catch (e: UnsupportedCharsetException) {
-                    return
-                }
+    private fun processResponseBody(response: Response, responseBody: ResponseBody, transaction: HttpTransaction) {
+        val contentType = responseBody.contentType()
+        val charset: Charset = contentType?.charset(UTF8) ?: UTF8
+        val contentLength = responseBody.contentLength()
+
+        val source = responseBody.source()
+        source.request(Long.MAX_VALUE) // Buffer the entire body.
+        var buffer = source.buffer()
+
+        if (CONTENT_ENCODING_GZIP.equals(response.headers()[CONTENT_ENCODING], ignoreCase = true)) {
```

Comment:
We already have this check in as `io.bodyIsGzipped()`. You could use it here.
However, I like you approach with constant. Could I ask you to modify `IOUtils` class as well to follow the same approach as here?